fds/utils.py

```diff
@@ -134,12 +134,64 @@ def get_input_from_user(question: str, type: str = "input") -> str:
     :param type: The type of input
     :return: output from the user
     """
-    questions = [
-        {
-            'type': type,
-            'message': question,
-            'name': 'question'
-        }
-    ]
-    answers = PyInquirer.prompt(questions)
-    return answers['question']
+    get_input = input
+    if type == "password":
+        get_input = getpass.getpass
+    display_text_to_user = f"{question}:"
+    answer = get_input(display_text_to_user)
+    if answer is None:
+        return get_input_from_user(question, type)
+    return answer
+
+
+def get_expand_input_from_user(question: str, choices: List[Dict[str, str]], default: Union[str, bool, int]) -> str:
+    """
+    Get expand input from the user
+    :param question: Question to ask the user
+    :param choices: List of choices from user
+    :param default: Default selected value
+    :return: user selected value
+    """
+    # Only pick choice keys
+    choice_keys = list(map(lambda x: x["key"], choices))
+    # Add an extra 'h' for help
+    choice_keys.append("h")
+    choices_string = "".join(choice_keys)
+    default_choice = list(filter(lambda x: x['value'] == default, choices))[0]
+    default_key = default_choice["key"]
+    display_text_to_user = f"{question}   ({choices_string}) [{default_key}]"
+    input_value = input(display_text_to_user) or default_choice["key"]
+    while input_value == 'h' or input_value == 'help':
+        # User chose help then we should show the list again with full choices
+        detailed_choices = list(map(lambda x: f"{x['key']}) {x['name']}", choices))
+        detailed_choices_string = "\n".join(detailed_choices)
+        display_text_to_user = f"{display_text_to_user}\n{detailed_choices_string} \nAnswer:"
+        input_value = input(display_text_to_user)
+    if input_value in choice_keys:
+        choice_made = list(filter(lambda x: x['key'] == input_value, choices))[0]
+        return choice_made["value"]
+    else:
+        print(f"Not a valid choice: please choose from the given choices ({choices_string})")
+        return get_expand_input_from_user(question, choices, default)
+
+
+def get_confirm_from_user(message: str, default: bool) -> bool:
+    """
+    Get the confirm response from user
+    :param message: Message to be shown to user
+    :param default: Default value
+    :return: User choice of confirm
+    """
+    choices = [{"key": "y", "value": True, "name": "Yes"}, {"key": "n", "value": False, "name": "No"}]
+    return bool(get_expand_input_from_user(message, choices, default))
+
+
+def get_list_choice_from_user(message: str, items_list: List[str]) -> str:
+    """
+    Get List choice from user
+    :param message: Message to be shown to user
+    :param list: List value
+    :return: User choice of list
+    """
+    choices = list(map(lambda x: {"key": items_list.index(x) + 1, "value": x, "name": x}, items_list))
+    return get_expand_input_from_user(message, choices, 1)
```

Comment:
Won't this display the following bad message to the user until they choose "h"?
`"{message} (123456h) [1]"
Maybe we need an additional optional flag to `get_expand_input_from_user` which tells it to always print the detailed list of options

Reply:
Well thats the functionality right now. I can change the display in a new PR if you want.